### PR TITLE
refactor(ValueCard): new design and precision fixes

### LINKED ISSUE #1141: [ValueCard] refactoring ideas
### Summary

This is a set of refactoring ideas to simplify the code of ValueCard.
We have so many similar props that we're passing through just to decide how to render low-down.  Need to move it up

1. Move off styled-components as the nesting is very confusing

ValueRenderer:
Self-measuring size improvement should allow us to remove unit, size, layout, isSmall, isMini, isVertical, external components are basically having to pass in parameters that calculate the font-size internally, but it's not explicit based on the component input what actual font size gets used

UnitRenderer:
Ideally he wouldn't need to know 'value', 'layout', 'isMini', 'attributeCount'

Attribute:
Why does he need to know the 'attributeCount', 'isMini', 'isSmall', 'size'.  Could attribute simply based on the measured width from SizeMe, dynamically figure out the appropriate font sizes to pass downstream to ValueRenderer and UnitRenderer
Add label to Attribute rendering. Merge AttributeWrapper being a styled-component into Attribute.  Sizing of the fonts is attribute's responsibility, labelSize, unitSize, valueSize up to Attribute to determine and we'll make Value, Unit and Label dumber and just take the sizes
 <AttributeWrapper
                      layout={layout}
                      isVertical={isVertical}
                      isSmall={attribute.secondaryValue !== undefined}
                      isMini={isMini}
                      size={newSize}
                      attributeCount={attributes.length}
                    >

ValueCard:
remove styled components as it's overcomplicating the render to understand them
Rename isVertical to showLabelAboveValue, 
layout determines are the attributes next to each other or above each other, rename attributeLayout
Remove isMini, isSmall determinations from ValueCard

**Additional context**

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

_NB: The core contributors will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_

### Available extra resources

What resources do you have to assist this effort?

_Carbon add-ons for Watson IoT is a collaborative system. We encourage teams to build components and submit them for integration._

### LINKED ISSUE #1276: [ValueCard] attribute layout rules
### Summary

This is an extension of the logic for the existing value cards (including outlook for other types in future).

The value cards have a number of issues open around them today on the topic of how they (at varying sizes) manage varying scales of content such as long values, long units, wrapping etc. This contribution puts forward a framework to think and talk about all of these elements, as well as a proposal for how we can address these needs.

**Additional context**
From [Internal Watson-IoT/pal-tracking#260](https://github.ibm.com/Watson-IoT/pal-tracking/issues/260)

> At a high level, we identify that a value card has a set X & Y space for its payload. Inside there will be the primary value along with supporting elements such as the unit, threshold alerts, variance, labels and more.
> 
> **The overall intended behavior to put it in a few points is:**
> -  The supporting elements will take up as much space as they require based on their content
> - The value will adjust its font size to balloon up or shrink down to fill any remaining space on a single line. 
> - If it is not possible to fit its full value on a single line at a 20pt, it will then begin to wrap.
> 
> This can be seen in more detail as such:
> Basic anatomy and scenarios
> 
> Supporting image 1: Value cards with multiple values.
> Multi-value cards
> 
> Supporting image 2: Examples of this logic applied to some cases with varying unit size and varying value length at different levels of complexity:
> Examples

### LINKED ISSUE #1892: [ValueCard] values not showing precision
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug

Precision can be set to 4, but the whole numbers are still being truncated. If precision is set, it should be assumed that the user wants to see all numbers

### To Reproduce

1. Go to ValueCard, medium size and add 2-3 attributes with precison: 4 and values > 1000
2. See that the value will be truncated to 1K

### Expected behavior

If I set precision to 4 I should be able to see 4 digits of information in whole numbers and decimal numbers.

### Environment/versions:

- OS: [e.g. MacOS, Windows]
- Browser: [e.g. chrome, safari]
- `carbon-addons-iot-react` version: [e.g. `v2.60.0`]

### Additional context

internal issue: https://github.ibm.com/wiotp/monitoring-dashboard/issues/1694

### PR DESCRIPTION
**Pull request by sstone2423**
Internal issue: https://github.ibm.com/wiotp/monitoring-dashboard/issues/1694
Closes #1892 
Closes #1141 
Part of #370 
Closes #1276 

**Summary**

*Note*: I would recommend not relying on the diff for the ValueCard files for this review as this is essentially a full rewrite. It will be much easier to read the files without the diff to see the new logic and review as if this was a new component being introduced.

- Redesign primarily to handle truncation and wrapping differently when precision is set
- Required a very large refactor as the old logic had very dense logic and conditionals around all of the different card sizes for truncating and wrapping

**Change List (commits, features, bugs, etc)**

- Removed styled components and swapped the dynamic variables for scss variables
- Removed `formatChartNumberWithPrecision` in favor of `formatNumberWithPrecision` as they had the same exact logic
- Refactored `formatNumberWithPrecision` to instead use the native I18n class to format numbers based on locale (may need to check other places in the code that we're using external libraries or our own logic to try and determine the locale format
- Extracted all attribute styling logic to Attribute to keep separation of concerns
- Renamed, removed, and refactored some of the stories to better reflect the possible use-cases

**Acceptance Test (how to verify the PR)**

1. ValueCard
  - ensure that each ValueCard story adheres to the new design spec seen in the internal issue above
  - Try using the font size knob to see that the font size responds correctly for the value
  - Try using the isNumberValueCompact knob and change the locale to see that it translates correctly

2. DashboardEditor
  - Add a new ValueCard, go to the settings tab, and try the font size number input and abbreviate numbers input. You should see both getting updated in the actions

### CODE REVIEW

**Review by scottdickerson — APPROVED** [review]
LGTM this is such a great refactor, I'm really glad to see it

File: src/components/ValueCard/_attribute.scss [hunk 1]
```diff
@@ -1,31 +1,55 @@
...
+    width: 100%;
+    // set the height so the height and value will always be in the same position regardless if they exist
+    height: 0.875rem;
+    margin-bottom: $spacing-02;
+  }
 
-.#{$iot-prefix}--value-card__attribute_trend-icon {
-  margin-right: $spacing-02;
+  &--secondary-value {
+    height: 24px;
```
Comment (davidicus) [inline src/components/ValueCard/_attribute.scss:47]: Can we use rems here?

File: src/components/ValueCard/_value-renderer.scss [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
-.#{$iot-prefix}--value-card__attribute-value--wrappable {
-  margin-right: $carbon--spacing-02;
-}
+.#{$iot-prefix}--value-card__value-renderer {
+  &--wrapper {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    display: flex;
+    //
+    // display: -webkit-box;
```
Comment (davidicus) [inline src/components/ValueCard/_value-renderer.scss:9]: Can we remove these styles?

File: src/utils/__tests__/__snapshots__/publicAPI.test.js.snap [hunk 3]
```diff
@@ -24755,7 +24772,6 @@ Map {
     "render": [Function],
   },
   "determineCardRange" => Object {},
-  "determineMaxValueCardAttributeCount" => Object {},
```
Comment (davidicus) [inline src/utils/__tests__/__snapshots__/publicAPI.test.js.snap:24758]: Is this a breaking change? or is this provided by the component?

**Review by davidicus — CHANGES_REQUESTED** [review]
Some minor comments but looks like a great refactor. Awesome!

File: src/utils/__tests__/__snapshots__/publicAPI.test.js.snap (hunk 3 above)
Comment (sstone2423) [reply src/utils/__tests__/__snapshots__/publicAPI.test.js.snap:24758]: Theres technically no 'max' number of attributes for ValueCard with the redesign, but it is a breaking change thats its no longer being exported. I can leave it in with a deprecation warning

File: src/components/ValueCard/_value-renderer.scss (hunk 2 above)
Comment (sstone2423) [reply src/components/ValueCard/_value-renderer.scss:9]: Yes, that was left in by mistake

File: src/utils/__tests__/__snapshots__/publicAPI.test.js.snap (hunk 3 above)
Comment (sstone2423) [reply src/utils/__tests__/__snapshots__/publicAPI.test.js.snap:24758]: I re-added the export and added a deprecation warning when in `__DEV__`

File: src/components/ValueCard/_value-renderer.scss (hunk 2 above)
Comment (sstone2423) [reply src/components/ValueCard/_value-renderer.scss]: Removed

File: src/components/ValueCard/_attribute.scss (hunk 1 above)
Comment (sstone2423) [reply src/components/ValueCard/_attribute.scss]: Swapped to rem

**Review by davidicus — APPROVED** [review]

**Pull request merged**